src/display/editor/annotation_editor_layer.js

```diff
@@ -404,6 +404,9 @@ class AnnotationEditorLayer {
         // Do nothing on right click.
         return;
       }
+
+      this.#uiManager._eventBus.dispatch("annotationstart", { source: this });
```

Comment:
*This comment applies elsewhere as well.*

 - Is there possibly a more "central" spot where this can be done, since having this code in a bunch of places seems potentially error prone (since it's easy to miss adding a case)?
 - The event name isn't very specific, as far as I'm concerned. (But please see below.)
 - Ideally the undo-bar hiding would, in my opinion, be placed within the `src/display/editor/`-code as much as possible. (With events only used for features that are clearly outside of that folder.)
   With that in mind, could we change this to the following instead?
   ```suggestion
         this.#uiManager._editorUndoBar?.hide()
   ```

Reply:
> Is there possibly a more "central" spot where this can be done, since having this code in a bunch of places seems potentially error prone (since it's easy to miss adding a case)?

We originally tried looking for a central place but couldn't find a good one. The problem is some of the editors are created when they are "enabled" by clicking on the toolbar (example: the ink one), while others are created only when the user actually starts writing in them (example: the freetext one).